vermouth/processors/molecule_grouper.py

```diff
@@ -0,0 +1,295 @@
+# Copyright 2020 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import collections
+import itertools
+import networkx as nx
+import numpy as np
+
+from ..selectors import is_water, selector_has_position
+from .processor import Processor
+from ..molecule import Molecule
+from ..log_helpers import StyleAdapter, get_logger
+
+LOGGER = StyleAdapter(get_logger(__name__))
+
+
+def expand_to_list(num_or_list, length):
+    """
+    If `num_or_list` is a single item, turns it into a list of length `length`.
+    Otherwise, returns `num_or_list`.
+
+    Parameters
+    ----------
+    num_or_list
+    lenght: int
+
+    Returns
+    -------
+    list
+    """
+    try:
+        len(num_or_list)
+    except TypeError:
+        num_or_list = [num_or_list] * length
+    return num_or_list
```

Comment:
Is this really needed. This function is only used twice in consecutive lines and very specific to the use case?

Reply:
Needed, no. But it's here now anyway, and it was little effort.